File: address-validation-us-3-dot-net-examples/GetBestMatchesSoapSdkExample.cs
using address_validation_us_3_dot_net.SOAP;

namespace address_validation_us_3_dot_net_examples
{
    internal static class GetBestMatchesSoapSdkExample
    {
        public static void Go(string LicenseKey, bool IsLive)
        { 
            Console.WriteLine("\r\n---------------------------------------------------");
            Console.WriteLine("Address Validation US 3 - GetBestMatches - SOAP SDK");
            Console.WriteLine("---------------------------------------------------");

            Console.WriteLine("\r\n* Input *\r\n");
            Console.WriteLine($"Business Name: ");
            Console.WriteLine($"Address 1    : 136 West Canon Perdido St");
            Console.WriteLine($"Address 2    : Suite D");
            Console.WriteLine($"City         : Santa Barbara");
            Console.WriteLine($"State        : CA");
            Console.WriteLine($"ZIP Code     : 93101");
            Console.WriteLine($"Is Live      : {IsLive.ToString()}");
            Console.WriteLine($"License Key  : {LicenseKey}");

            GetBestMatchesValidation getBestMatchesValidation = new(IsLive);
            AV3Service.BestMatchesResponse response = getBestMatchesValidation.GetBestMatches(
                "",
                "136 West Canon Perdido St",
                "Suite D",
                "Santa Barbara",
                "CA",
                "93101",
                LicenseKey
            ).Result;

            if (response.Error is null)
            {
                Console.WriteLine("\r\n* Validation *\r\n");

                foreach (AV3Service.Address address in response.Addresses)
                {
                    Console.WriteLine($"Address 1          : {address.Address1}");
                    Console.WriteLine($"Address 2          : {address.Address2}");
                    Console.WriteLine($"City               : {address.City}");
                    Console.WriteLine($"State              : {address.State}");
                    Console.WriteLine($"ZIP+4              : {address.Zip}");
                    Console.WriteLine($"Is Residential     : {address.IsResidential}");
                    Console.WriteLine($"DPV                : {address.DPV}");
                    Console.WriteLine($"DPV Desc           : {address.DPVDesc}");
                    Console.WriteLine($"DPV Notes          : {address.DPVNotes}");
                    Console.WriteLine($"DPV Notes Desc     : {address.DPVNotesDesc}");
                    Console.WriteLine($"Corrections        : {address.Corrections}");
                    Console.WriteLine($"Corrections Desc   : {address.CorrectionsDesc}");
                    Console.WriteLine($"Barcode Digits     : {address.BarcodeDigits}");
                    Console.WriteLine($"Carrier Route      : {address.CarrierRoute}");
                    Console.WriteLine($"Congress Code      : {address.CongressCode}");
                    Console.WriteLine($"County Code        : {address.CountyCode}");
                    Console.WriteLine($"County Name        : {address.CountyName}");
                    Console.WriteLine($"Fragment House     : {address.FragmentHouse}");
                    Console.WriteLine($"Fragment Pre Dir   : {address.FragmentPreDir}");
                    Console.WriteLine($"Fragment Street    : {address.FragmentStreet}");
                    Console.WriteLine($"Fragment Suffix    : {address.FragmentSuffix}");
                    Console.WriteLine($"Fragment Post Dir  : {address.FragmentPostDir}");
                    Console.WriteLine($"Fragment Unit      : {address.FragmentUnit}");
                    Console.WriteLine($"Fragment           : {address.Fragment}");
                    Console.WriteLine($"Fragment PMB Prefix: {address.FragmentPMBPrefix}");
                    Console.WriteLine($"Fragment PMB Number: {address.FragmentPMBNumber}");
                }
                Console.WriteLine($"Is CASS: {response.IsCASS}");
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");

                Console.WriteLine($"Error Type     : {response.Error.Type}");
                Console.WriteLine($"Error Type Code: {response.Error.TypeCode}");
                Console.WriteLine($"Error Desc     : {response.Error.Desc}");
                Console.WriteLine($"Error Desc Code: {response.Error.DescCode}");
            }
        }
    }
}

File: address-validation-us-3-python-examples/get_best_matches_soap_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-validation-us-3-python/SOAP"))

from get_best_matches_soap import GetBestMatchesValidation

def get_best_matches_soap_sdk_go(license, is_live_key):
    print("\n" + "-" * 51)
    print("Address Validation US 3 - GetBestMatches - SOAP SDK")
    print("-" * 51)

    print("\n* Input *\n")
    print(f"Business Name: ")
    print(f"Address 1    : 136 West Canon Perdido St")
    print(f"Address 2    : Suite D")
    print(f"City         : Santa Barbara")
    print(f"State        : CA")
    print(f"ZIP Code     : 93101")
    print(f"Is Live      : {is_live_key}")
    print(f"License Key  : {license}")

    try:
        service = GetBestMatchesValidation(license, is_live=is_live_key, timeout_ms=10000)
        response = service.get_best_matches(
            business_name="",
            address1="136 West Canon Perdido St",
            address2="Suite D",
            city="Santa Barbara",
            state="CA",
            postal_code="93101"
        )

        if not hasattr(response, 'Error'):
            print("\n* Validation *\n")
            for address in response.Addresses[0]:
                print(f"Address 1          : {address.Address1}")
                print(f"Address 2          : {address.Address2}")
                print(f"City               : {address.City}")
                print(f"State              : {address.State}")
                print(f"ZIP+4              : {address.Zip}")
                print(f"Is Residential     : {address.IsResidential}")
                print(f"DPV                : {address.DPV}")
                print(f"DPV Desc           : {address.DPVDesc}")
                print(f"DPV Notes          : {address.DPVNotes}")
                print(f"DPV Notes Desc     : {address.DPVNotesDesc}")
                print(f"Corrections        : {address.Corrections}")
                print(f"Corrections Desc   : {address.CorrectionsDesc}")
                print(f"Barcode Digits     : {address.BarcodeDigits}")
                print(f"Carrier Route      : {address.CarrierRoute}")
                print(f"Congress Code      : {address.CongressCode}")
                print(f"County Code        : {address.CountyCode}")
                print(f"County Name        : {address.CountyName}")
                print(f"Fragment House     : {address.FragmentHouse}")
                print(f"Fragment Pre Dir   : {address.FragmentPreDir}")
                print(f"Fragment Street    : {address.FragmentStreet}")
                print(f"Fragment Suffix    : {address.FragmentSuffix}")
                print(f"Fragment Post Dir  : {address.FragmentPostDir}")
                print(f"Fragment Unit      : {address.FragmentUnit}")
                print(f"Fragment           : {address.Fragment}")
                print(f"Fragment PMB Prefix: {address.FragmentPMBPrefix}")
                print(f"Fragment PMB Number: {address.FragmentPMBNumber}")
            print(f"Is CASS: {response.IsCASS}")
        else:
            print("\n* Error *\n")
            err = response.Error
            print(f"Error Type     : {err.Type}")
            print(f"Error Type Code: {err.TypeCode}")
            print(f"Error Desc     : {err.Desc}")
            print(f"Error Desc Code: {err.DescCode}")
    except RuntimeError as e:
        print("Error calling service:", e)
